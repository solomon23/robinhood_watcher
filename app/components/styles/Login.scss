.btn, .input {
  width: 200px;
  height: 26px;
  padding-left: 8px;
  border-radius: 4px;
  display: inline-block;
  margin-bottom: 8px;
}

.login {
  padding-top: 16px;
  text-align: center;
}

.btn {
  background-color: #21ce99;
  padding: 0;
  border-style: none;
  color: #fff;
  width: 211px;
  height: 30px;
  font-weight: bold;
}

.loading {
  opacity: 0.5;
  pointer-events: none;
  cursor: default;
}
