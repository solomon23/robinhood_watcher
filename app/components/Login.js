// @flow
import React, { Component } from 'react'
import classnames from 'classnames'
import styles from './styles/Login.scss'

type Props = {
  user: object,
  login: () => null
}

export default class Login extends Component<Props> {
  props: Props

  constructor(props, context) {
    super(props, context)

    this.login = this.login.bind(this)

    this.state = {
      token: '',
      username: '',
      password: '',
    }
  }

  login() {
    const { login } = this.props
    const { username, password, token } = this.state

    login(username, password, token || null)
  }

  oChangeField(field, e) {
    this.setState({ [field]: e.target.value })
  }

  render() {
    const { user } = this.props
    const { username, token, password } = this.state

    return (
      <div className={classnames(styles.login, { [styles.loading]: user.isLoading })}>
        <input
          type="text"
          className={styles.input}
          placeholder="USERNAME"
          value={username}
          onChange={e => this.oChangeField('username', e)}
        />
        <input
          className={styles.input}
          type="password"
          placeholder="PASSWORD"
          value={password}
          onChange={e => this.oChangeField('password', e)}
        />
        {(user.needsMfa || token) && (
          <input
            type="text"
            className={styles.input}
            placeholder="TOKEN"
            value={token}
            onChange={e => this.oChangeField('token', e)}
          />
        )}
        <button
          className={styles.btn}
          onClick={this.login}
          data-tclass="btn"
          type="button"
        >
          LOGIN
        </button>
      </div>
    )
  }
}
