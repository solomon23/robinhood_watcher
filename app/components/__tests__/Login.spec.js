import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { shallowToJson } from 'enzyme-to-json'
import Login from '../Login'

Enzyme.configure({ adapter: new Adapter() })

let actions = null

const defaultState = {
  user: {},
}

describe('Login component', () => {
  beforeEach(() => {
    actions = { login: jest.fn() }
  })

  it('should display', () => {
    const component = shallow(<Login {...defaultState} {...actions} />)
    expect(shallowToJson(component)).toMatchSnapshot()
  })

  it('should display loading', () => {
    const state = { ...defaultState, user: { isLoading: true } }
    const component = shallow(<Login {...state} {...actions} />)
    expect(shallowToJson(component)).toMatchSnapshot()
  })

  it('should login', () => {
    const component = shallow(<Login {...defaultState} {...actions} />)
    component.find('input').at(0).simulate('change', { target: { value: 'user' } })
    component.find('input').at(1).simulate('change', { target: { value: 'password' } })
    component.find('.btn').simulate('click')
    expect(actions.login).toHaveBeenCalledWith('user', 'password', null)
  })

  it('should show token', () => {
    const state = { ...defaultState, user: { needsMfa: true } }
    const component = shallow(<Login {...state} {...actions} />)
    expect(shallowToJson(component)).toMatchSnapshot()
  })

  it('should send token', () => {
    const state = { ...defaultState, user: { needsMfa: true } }
    const component = shallow(<Login {...state} {...actions} />)
    component.find('input').at(0).simulate('change', { target: { value: 'user' } })
    component.find('input').at(1).simulate('change', { target: { value: 'password' } })
    component.find('input').at(2).simulate('change', { target: { value: 'token' } })
    component.find('.btn').simulate('click')
    expect(actions.login).toHaveBeenCalledWith('user', 'password', 'token')
  })
})
