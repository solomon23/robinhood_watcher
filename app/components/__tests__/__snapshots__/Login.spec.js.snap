// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should display 1`] = `
<div
  className="login"
>
  <input
    className="input"
    onChange={[Function]}
    placeholder="USERNAME"
    type="text"
    value=""
  />
  <input
    className="input"
    onChange={[Function]}
    placeholder="PASSWORD"
    type="password"
    value=""
  />
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    LOGIN
  </button>
</div>
`;

exports[`Login component should display loading 1`] = `
<div
  className="login loading"
>
  <input
    className="input"
    onChange={[Function]}
    placeholder="USERNAME"
    type="text"
    value=""
  />
  <input
    className="input"
    onChange={[Function]}
    placeholder="PASSWORD"
    type="password"
    value=""
  />
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    LOGIN
  </button>
</div>
`;

exports[`Login component should show token 1`] = `
<div
  className="login"
>
  <input
    className="input"
    onChange={[Function]}
    placeholder="USERNAME"
    type="text"
    value=""
  />
  <input
    className="input"
    onChange={[Function]}
    placeholder="PASSWORD"
    type="password"
    value=""
  />
  <input
    className="input"
    onChange={[Function]}
    placeholder="TOKEN"
    type="text"
    value=""
  />
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    LOGIN
  </button>
</div>
`;
